giraffe/src/components/SimpleTable/PagedTable.tsx

```diff
@@ -112,13 +123,13 @@ const subsetResult = (
       })
     )
     .filter(column => !!column.data.filter(_c => _c !== undefined).length)
-    .reduce((arr, curr) => {
-      if (arr[curr.name]) {
-        arr[curr.name].push(curr)
-        return arr
+    .reduce((acc, curr) => {
```

Comment:
It's astounding, bordering on impressive how bad this code is. Between the ridiculous variable names, the unnecessary `map().filter().reduce()` pattern that could be combined into a single loop that is much conceptually simpler, and the sheer number of loops in this function, which is on a hot rendering path, it's just a complete mess. I bet that simplifying this function will increase performance significantly, cause it is pretty bad.

Let me know if you want help with it. No one should have to deal with this code alone.

Reply:
The whole sequence of array functions start from `columns` inside of the `table` property which is the object representing all of the data. The `columns` are not that many. They `can` be many if cardinality is high, if I am understanding cardinality correctly. But on the other hand, with a lot of columns, comes a lot of scrolling horizontally for the user to see all of the columns. Likely, a user would not want to visualize something that requires him/her to scroll incessantly to the side just to see some data.

So, the amount of performance to be gained here would depend on the type of data the user wants to see. I would in most cases, the number of columns here don't make enough of a difference on the rendering performance.

We can certainly and should change this code to make it more readable. Using one pass through the columns is a good approach. But also, there's a lot of work being done on each pass. We will need to be clear on the steps to make it easy to follow.

Let's leave this part for a future change. This is not the last update to Simple Table.